src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Connection.cs

```diff
@@ -156,9 +164,19 @@ public virtual void OnSocketClosed()
         }
 
         // Called on Libuv thread
-        public void Tick()
+        public void Tick(long timestamp)
         {
-            _frame.Tick();
+            if (_timeoutEnabled)
+            {
+                if (_millisecondsSinceLastConnectionEvent > ServerOptions.Limits.ConnectionTimeout.TotalMilliseconds)
```

Comment:
Store this so ms doesn't need to be recalculated. You could also add a second to the configuration since `_millisecondsSinceLastConnectionEvent` could be up to a second too large.